infra.go

```diff
@@ -119,19 +101,20 @@ func (i *InfrastructureResource) ResourceType() TaggableResource {
 }
 
 func (client *Client) CreateInfrastructure(input InfraInput) (*InfrastructureResource, error) {
+	data := JSON(input.Data)
 	i := InfrastructureResourceInput{
 		Schema: &InfrastructureResourceSchemaInput{Type: input.Schema},
-		Data:   input.Data,
+		Data:   &data,
```

Comment:
why can we not use `&input.Data` here?

Reply:
There's a hardcoded rule in gen.go where `Data` always marshals to `JSON`.